tests/api/rbac.test.js

```diff
@@ -50,64 +62,125 @@ describe(`[P2][Sev2][${squad}] Search API: Verify RBAC`, () => {
+    test('should receive ConfigMap', () => ValidateSearchData({ user, kind: 'configmap' }), validationTimeout)
+    test('should receive Node', () => ValidateSearchData({ user, kind: 'node' }), validationTimeout)
   })
 
-  describe(`with user ${usr3} (all resources in namespace ${ns})`, () => {
-    test.todo('should validate RBAC configuration.')
-    test.todo('should validate results from search.')
+  describe(`with user ${usr3} (admin for namespace ${ns})`, () => {
+    beforeAll(async () => {
+      user = await getUserContext({ usr: usr3, ns, retryWait: 4000 })
+    })
+
+    test('should validate RBAC configuration for user', () => {
+      expectCli(`oc auth can-i list secret -n ${ns} --as=${user.fullName}`).not.toThrow()
+      expectCli(`oc auth can-i list configmap -n ${ns} --as=${user.fullName}`).not.toThrow()
+      expectCli(`oc auth can-i list configmap -n default --as=${user.fullName}`).toThrow()
+      expectCli(`oc auth can-i list node --as=${user.fullName}`).toThrow()
+    })
+
+    test('should receive Secret', () => ValidateSearchData({ user, kind: 'secret' }), validationTimeout)
+    test('should receive ConfigMap', () => ValidateSearchData({ user, kind: 'configmap' }), validationTimeout)
+    test('should not receive Node', () => ValidateSearchData({ user, kind: 'node' }), validationTimeout)
+
+    /* FIXME: Keeping this test disabled because user is also getting authorized to get ConfigMaps in
+     * open-cluster-management. This is coming from Kubernetes, not search.
+    test(`should not match resources from other namespaces`, async () => {
+      expect(() => execSync(`oc auth can-i list configmap -n open-cluster-management --as=${user.fullName}`)).toThrow()
+      const q = searchQueryBuilder({ filters: [{ property: 'kind', values: ['configmap'] }] })
+      const res = await sendRequest(q, user.token)
+      const items = res.body.data.searchResult[0].items
+
+      expect(items.find(({ namespace }) => namespace.toLowerCase() !== ns)).toEqual(undefined)
+    })
+    */
   })
 
-  describe(`with user search-user4 (access to deployment but not pod)`, () => {
-    test.todo('should validate RBAC configuration.')
+  describe(`with user ${usr4} (access to deployment but not pod)`, () => {
+    beforeAll(async () => {
+      user = await getUserContext({ usr: usr4, ns, retryWait: 4000 })
+    })
+
+    test('should validate RBAC configuration.', () => {
+      expectCli(`oc auth can-i list secret -n ${ns} --as=${user.fullName}`).toThrow()
+      expectCli(`oc auth can-i list pod -n ${ns} --as=${user.fullName}`).toThrow()
+      expectCli(`oc auth can-i list deployment -n ${ns} --as=${user.fullName}`).not.toThrow()
+    })
+
+    test(`should not get Secret`, () => ValidateSearchData({ user, kind: 'secret', namespace: ns }), validationTimeout)
+    test('should not get Pod', () => ValidateSearchData({ user, kind: 'configmap', namespace: ns }), validationTimeout)
+    test(
+      'should get Deploymt',
```

Comment:
```suggestion
      'should get Deployment',
```

Deployment is spelled wrong.

Reply:
This was on purpose trying to keep it in a single line, but it went over anyways. :-(